xena-txrx.py

```diff
@@ -89,13 +89,31 @@ def main():
         print('Xena chassis connection successful')
     except Exception as e:
         # failure to connect returns object 'timeout', triggering a TypeError
-        print('Error: connection to Xena chassis failed;', e)
-    else:
-        # executes if no exception
-        print('Disconnecting from Xena chassis...')
-        del xsocket
+        print('Error: connection to Xena chassis failed; ', e)
+
+    # create the manager session
+    xm = XenaManager(xsocket, xenaUsr, xenaPwd)
+
+    try:
```

Comment:
For the finally statement below to work correctly, lets wrap the port 0 and port 1 into a single try statement.  If we hit a problem with the first port try block it appears the finally block below will not get executed.

Reply:
Combined the port try statements and tested - finally appears to now execute correctly.